src/hooks/useBasicTokens.ts

```diff
@@ -0,0 +1,172 @@
+import {
+  ChainId,
+  PoolTypes,
+  SDL_WETH_SUSHI_LP_CONTRACT_ADDRESSES,
+  VOTING_ESCROW_CONTRACT_ADDRESS,
+} from "../constants"
+import {
+  MulticallCall,
+  MulticallContract,
+  MulticallProvider,
+} from "../types/ethcall"
+import { UseQueryResult, useQuery } from "@tanstack/react-query"
+import { chunkedTryAll, isSynthAsset } from "../utils"
+
+import { BasicPoolsContext } from "../providers/BasicPoolsProvider"
+import { Contract } from "ethcall"
+import ERC20_ABI from "../constants/abis/erc20.json"
+import { Erc20 } from "./../../types/ethers-contracts/Erc20.d"
+import { GaugeContext } from "../providers/GaugeProvider"
+import { MinichefContext } from "../providers/MinichefProvider"
+import { areGaugesActive } from "../utils/gauges"
+import { useActiveWeb3React } from "../hooks"
+import { useContext } from "react"
+import { useMulticallProvider } from "./useMulticallProvider"
+
+export type BasicToken = {
+  address: string
+  name: string
+  symbol: string
+  decimals: number
+  isLPToken: boolean
+  isSynthetic: boolean
+  typeAsset: PoolTypes
+}
+export type BasicTokens = Partial<{ [address: string]: BasicToken }> | null
+
+export const useBasicTokens = (): UseQueryResult<BasicTokens> => {
+  const { chainId } = useActiveWeb3React()
+  const basicPools = useContext(BasicPoolsContext)
+  const minichefData = useContext(MinichefContext)
+  const { gauges } = useContext(GaugeContext)
+  const { data: ethCallProvider } = useMulticallProvider()
+  const gaugesAreActive = areGaugesActive(chainId)
+  const lpTokens = new Set()
+  const tokenType: Partial<{ [tokenAddress: string]: PoolTypes }> = {}
+
+  return useQuery(["tokens"], async () => {
+    if (!chainId || !basicPools || !ethCallProvider) {
+      return null
+    }
+    const targetTokenAddresses = new Set(
+      Object.values(basicPools)
+        .map((pool) => {
```

Comment:
I feel I was probably the one who wrote the code originally, but I'd prefer not to have side effects in map or reduce blocks - forEach is fine.